helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/CodeTransformations.java

```diff
@@ -304,6 +314,59 @@ public static String transformParseCalls(String code, AvroVersion minSupportedVe
     return prefix + newParseCall + restOfCode;
   }
 
+  /**
+   * escapes any control characters inside string literals that are part of an argument json
+   * PACKAGE PRIVATE FOR TESTING
+   * @param str a piece of json, as a java string literal
+   * @return the given piece of json, safe for use as a java string literal
+   */
+  static String escapeJavaLiteral(String str) {
+    if (str == null || str.isEmpty()) {
+      return str;
+    }
+
+    int quotedStrStart; //start of current quoted string (index of the starting quote character)
+    int quotedStrEnd;   //end of current quoted string (index of the end quote character)
+    int lastEnd = 0;    //end of previous quoted string (index of the end quote character)
+    StringBuilder result = new StringBuilder(str.length());
+
+    quotedStrStart = str.indexOf("\\\"", lastEnd);
+    while (quotedStrStart >= 0) {
+      quotedStrEnd = quotedStrStart;
+      while (true) {
+        quotedStrEnd = str.indexOf("\\\"", quotedStrEnd + 2);
+        if (quotedStrEnd < 0) {
+          throw new IllegalArgumentException(
+              "unterminated string literal starting at offset " + quotedStrStart + " in " + str);
+        }
+        char precedingChar = str.charAt(quotedStrEnd - 1); //character before the closing quote
+        if (precedingChar != '\\') {
+          break;
+        }
+      }
+
+      String quotedStr = str.substring(quotedStrStart + 2, quotedStrEnd); //without enclosing quotes
+
+      //we assume we start with a properly escaped json inside, so we double all backslashes, then take one out
+      //if it was escaping a quote :-)
+      String escaped = quotedStr.replace("\\", "\\\\").replace("\\\"", "\"");
```

Comment:
Since Java escaping is kind of obtuse, could we ease readability by replacing those strings with constants? Something like:

```
static final String BACKSLASH = "\\";
static final String DOUBLE_BACKSLASH = "\\\\";
static final String QUOTE = "\"";
static final String BACKSLASH_QUOTE = "\\\"";
```